_blog/scaling-dot-net-core-microservices/index.md

```diff
@@ -0,0 +1,403 @@
+If you have a console application that is processing a lead, how do you manage this when the orchestrator fires the `Ctrl + C` event?
+Do you delete the process or wait for it to complete?  These are the decisions you should be making prior to moving to a container.
+Also, this will empower your team later on, as you'll know how to manage when you have an actual fire.
+
+### Docker Build
+
+We use Docker build to create Docker images.  Docker images can then be used to spin up containers.
+
+> TOP TIP: Environment Variables are your Best Friend Forever (BFF)
+
+When running containers, you have the option to pass variables into the container via a number of ways:
+
+* Volumes
+* Environment Variable
+
+When should you them?  Use volumes for static assets that might need to change.  For example, if you have a website then load your images via a volume, or if you have a database then keep the data within a volume so that you don't lose it when the container is deleted.
+Use Environment Variables for connections strings and any secrets that you're managing with the application.
+
+### Docker Compose
+
+When you have lots of images that need to run together then one option is Docker Compose.  By creating a `docker-compose` file, you can run a command to spin up a local environment within a terminal window and then access the services locally.
+
+## Orchestrator to the rescue
+
+So far, everything we've spoken about has been based on a single computer.  Now I can guess what you're thinking...
+
+> But if all my code is running in production on one box, what happens when that dies?  
+> *Everyone, all the time.*
+
+If you're dependent on any one thing then you're getting ready for a bad time.
+Your application, infrastructure and sanity can only cope with so much.
+Luckily, Orchestrator's like Kubernetes are making this simpler for you to manage.
+Orchestrator's are based on a declarative state, this means you declare of how much you should have and Kubernetes will maintain it.
+So if I want four instances of my website running, I'll get four.  If one instance dies another one will spin up.
+If another instance comes back to life and I have five websites running, then Kubernetes will send a kill command to one of the instances and get it back to four.
+This happens 24/7, without human interaction allowing you to stay asleep and enjoy your life away from a computer.
+
+### Setup the Application
+
+We pass `yaml` files to Kubernetes via `kubectl` to declare what state we want our application to run within.
+Lets get the application up and running, then we'll look into a solution for our scaling issue.
+
+### Minikube Vs Cloud
+
+Minikube creates a local Kubernetes cluster on your computer that you can manage yourself.
+How much does Minikube cost?  Nothing!  You're running it locally so this is one less thing to worry about.
+You shouldn't use Minikube to manage an actual deployment, I'd love to see a blog post about it.
+
+### Running in the Cloud
```

Comment:
I think you're name dropping again. You explained what minikube is briefly and now you're jumping to cloud. What happened to minikube? Should I install it or move on?
If it is not necessary, then don't tell me about it.